test/github-issues/3158/entity/SessionSettings.ts

```diff
@@ -0,0 +1,13 @@
+import { Entity, OneToOne, JoinColumn } from "../../../../src";
+import { Session } from "./Session";
+
+@Entity({
+    name: "SessionSettings"
+})
+export class SessionSettings  {
+
+    @OneToOne(type => Session, session => session.id, { primary: true})
```

Comment:
btw just want to warn you that in `0.3.0` defining a primary column (.e.g. sessionId) is required for primary keyed relations now